tests/protocol/mock_wire_protocol.py

```diff
@@ -28,48 +28,56 @@ def create(mock_wire_data_file):
     Creates a mock WireProtocol object that will return the data specified by 'mock_wire_data_file' (which must
     follow the structure of the data files defined in tests/protocol/mockwiredata.py).
 
-    NOTE: This function creates mocks for azurelinuxagent.common.utils.restutil.http_get and
+    NOTE: This function creates mocks for azurelinuxagent.common.utils.restutil.http_request and
           azurelinuxagent.common.protocol.wire.CryptUtil. These mocks can be stopped using the
           methods stop_mock_http_get() and stop_mock_crypt_util().
 
     The return value is an instance of WireProtocol augmented with these properties/methods:
 
         * mock_wire_data - the WireProtocolData constructed from the mock_wire_data_file parameter.
-        * stop_mock_http_get() - stops the mock for restutil.http_get
+        * stop_mock_http_request() - stops the mock for restutil.http_request
         * stop_mock_crypt_util() - stops the mock for CrypUtil
+        * stop() - stops both mocks
     """
-    def stop_mock_http_get():
-        if stop_mock_http_get.mock is not None:
-            stop_mock_http_get.mock.stop()
-            stop_mock_http_get.mock = None
-    stop_mock_http_get.mock = None
+    def stop_mock_http_request():
+        if stop_mock_http_request.mock is not None:
+            stop_mock_http_request.mock.stop()
+            stop_mock_http_request.mock = None
+    stop_mock_http_request.mock = None
 
     def stop_mock_crypt_util():
         if stop_mock_crypt_util.mock is not None:
             stop_mock_crypt_util.mock.stop()
             stop_mock_crypt_util.mock = None
     stop_mock_crypt_util.mock = None
 
+    def stop():
+        stop_mock_crypt_util()
+        stop_mock_http_request()
+
     protocol = WireProtocol(KNOWN_WIRESERVER_IP)
     protocol.mock_wire_data = WireProtocolData(mock_wire_data_file)
-    protocol.stop_mock_http_get = stop_mock_http_get
+    protocol.stop_mock_http_request = stop_mock_http_request
     protocol.stop_mock_crypt_util = stop_mock_crypt_util
+    protocol.stop = stop
```

Comment:
those stop functions are mainly for debugging purposes; they come handy when there are conflicts with the mocks used by tests and the ones used here.